app/models/network_port.rb

```diff
@@ -6,15 +6,17 @@ class NetworkPort < ApplicationRecord
   # "ManageIQ::Providers::NetworkManager"
   belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "ManageIQ::Providers::BaseManager"
   belongs_to :cloud_tenant
-  belongs_to :cloud_subnet
   belongs_to :device, :polymorphic => true
 
   has_and_belongs_to_many :security_groups
 
-  has_one :network_router, :through => :cloud_subnet
   has_one :floating_ip
   # TODO(lsmola) can this really happen? If not remove it
   has_many :floating_ips
+  has_many :cloud_subnet_network_ports
+  has_many :cloud_subnets, :through => :cloud_subnet_network_ports
```

Comment:
Also don't think you need the intermediate model and just use a normal `has_and_belongs_to_many`.